processor/src/errors.rs

```diff
@@ -28,3 +30,11 @@ pub enum ExecutionError {
     SyscallTargetNotInKernel(Digest),
     UnexecutableCodeBlock(CodeBlock),
 }
+
+impl Display for ExecutionError {
+    fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(fmt, "{self:?}")
+    }
+}
+
+impl Error for ExecutionError {}
```

Comment:
This should probably be gated by `std` feature (see [here](https://github.com/0xPolygonMiden/miden-vm/blob/next/assembly/src/errors.rs#L133) for an example).